### ISSUE TITLE
[iOS] ListViewItem inside a horizontal ListView may not render completely at first under certain conditions

### ISSUE THREAD

**Issue by takla21** [unoplatform/uno#12854]
### Current behavior

![ezgif com-video-to-gif](https://github.com/unoplatform/uno/assets/17461593/b2378c41-00ec-4220-9879-5e144d44ab73)

At first, there is nothing in the bottom part of the item. However, once the user selects an item in `ListView`, a white `Rectangle` (and a green one for the selected item) appears

The bug occurs under the following conditions:
- Retrieving `ItemsPresenter` from `ListView` and setting its `MinHeight` after it has been loaded (with a delay).
```cs
private void ListViewContainer_DataContextChanged(FrameworkElement sender, DataContextChangedEventArgs args)
    {
        _dataContextLoaded = true;
        if (_isLoaded)
        {
            _ = SetMinHeight();
        }
    }

    protected override void OnApplyTemplate()
    {
        base.OnApplyTemplate();
        _isLoaded = true;
        if (_dataContextLoaded)
        {
            _ = SetMinHeight();
        }
    }

   private async Task SetMinHeight()
    {
        await Task.Delay(100);
        _itemsPresenter!.MinHeight = 800;
    }
```
- Having a horizontal `ListView` nested within a vertical `ListView`.
- Applying a height to `ItemsContainerStyle` of the horizontal `ListView`.
- Using a StackPanel (or any custom class that inherits it) as the ItemsPanelTemplate for the horizontal `ListView`.
```xaml
<Setter Property="ItemsPanel">
		<Setter.Value>
			<ItemsPanelTemplate>
				<StackPanel Orientation="Horizontal" />
			</ItemsPanelTemplate>
		</Setter.Value>
	</Setter>
```

### Expected behavior

![winuiselec](https://github.com/unoplatform/uno/assets/17461593/c95d7694-68bb-4f37-9509-b138426f4a9c)

On the other platforms used on my project (Android. WinUI), we should always see the white/green line. Even with the condition specify above.

### How to reproduce it (as minimally and precisely as possible)

- Download this sample [ListViewItemSelected.zip](https://github.com/unoplatform/uno/files/12009338/ListViewItemSelected.zip).
- Deploy on iOS.
- Click on one of the items.

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

5.0.0-dev.1303

### Affected platforms

iOS

### IDE

Visual Studio 2022

### IDE version

17.6.1

### Anything else we need to know?

In the sample provided above, here is some guidance:

- `ListViewContainer` is a custom control that simulates the delay when manipulating the MinHeight property of the `ItemsPresenter` in the horizontal `ListView`.
- In the Helper.cs file, you will find extension methods that allow you to search for any item in the visual tree. These methods are useful for retrieving the `ItemsPresenter` from the vertical `ListView`.
- `<Style x:Key="ListViewContainerStyle"`  defines the style for `ListViewContainer`, which contains the vertical `ListView`.
- `<DataTemplate x:Key="TitlesTemplate">` represents the `DataTemplate` used for the vertical `ListView` that contains the horizontal `ListView`.
- `<Style x:Key="TitleListViewItemStyle"` specifies the `ItemsContainerStyle` for the horizontal `ListView`, setting a specific height.
- `<Style x:Key="HorizontalListViewStyle"` defines the style for the horizontal ListView."

**Comment by ajpinedam** [comment]
@takla21 

For the Scrollerviewer could you test changing these properties on the `HorizontalListViewStyle` Style.

```
  VerticalScrollMode="Disabled"
  VerticalScrollBarVisibility="Hidden"
  IsVerticalScrollChainingEnabled="False"
  IsVerticalRailEnabled="False"
```

Why?

This creates a double vertical scroller which, for iOS, we always need to avoid. I see that the vertical Scrolling on this Style is not needed for this specific case. 

Give it a try, and let me know if this works for you.

The full Style would look like this:

```xml
<Style x:Key="HorizontalListViewStyle"
		TargetType="ListView">
	<Setter Property="ItemsPanel">
		<Setter.Value>
			<ItemsPanelTemplate>
				<StackPanel Orientation="Horizontal" />
			</ItemsPanelTemplate>
		</Setter.Value>
	</Setter>

	<Setter Property="Template">
		<Setter.Value>
			<ControlTemplate TargetType="ListView">
				<ScrollViewer x:Name="ScrollViewer"
						xamarin:Style="{StaticResource ListViewBaseScrollViewerStyle}"
						TabNavigation="{TemplateBinding TabNavigation}"
						HorizontalScrollMode="Enabled"
						HorizontalScrollBarVisibility="Hidden"
						IsHorizontalScrollChainingEnabled="True"
						IsHorizontalRailEnabled="True"
						VerticalScrollMode="Disabled"
						VerticalScrollBarVisibility="Hidden"
						IsVerticalScrollChainingEnabled="False"
						IsVerticalRailEnabled="False"
						ZoomMode="{TemplateBinding ScrollViewer.ZoomMode}"
						IsDeferredScrollingEnabled="{TemplateBinding ScrollViewer.IsDeferredScrollingEnabled}"
				                BringIntoViewOnFocusChange="{TemplateBinding ScrollViewer.BringIntoViewOnFocusChange}"
						AutomationProperties.AccessibilityView="Raw">
					<ItemsPresenter Header="{TemplateBinding Header}"
							HeaderTemplate="{TemplateBinding HeaderTemplate}"
							HeaderTransitions="{TemplateBinding HeaderTransitions}"
							Footer="{TemplateBinding Footer}"
							FooterTemplate="{TemplateBinding FooterTemplate}"
							FooterTransitions="{TemplateBinding FooterTransitions}"
							Padding="{TemplateBinding Padding}" />
				</ScrollViewer>
			</ControlTemplate>
		</Setter.Value>
	</Setter>
</Style>
```

**Comment by takla21** [comment]
@ajpinedam So I tried what you suggested in on our app and it does not fix our issue even though it does in the sample. I'll have to compare what else is missing in between those two projects.

**Comment by ajpinedam** [comment]
Thanks @takla21 . Let me know what you find and if there's anything I can do to help.

**Comment by ajpinedam** [comment]
@takla21  did you find the reason why it's not working on the app? Do you need support on that?

**Comment by takla21** [comment]
@ajpinedam Sorry we didn't have time to focus on this unfortunetly

**Comment by jeromelaban** [comment]
A possible workaround for this issue could be to use an `ItemsRepeater` and [ItemsRepeater extensions](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/helpers/itemsrepeater-extensions.html) for selection.

**Comment by kazo0** [comment]
@ajpinedam can we take another look at this with the sample app that was provided? 

I'm assuming just update it to latest 4.x and see if we can repro/fix?

**Comment by ajpinedam** [comment]
Hey @kazo0 thanks for the follow-up.

The issue was not really an issue on Uno, so there was nothing to be fixed. The problem was found to be a custom Template for a control that included a ScrollView in the same orientation as an Outter ScrollView. For iOS, this is a no-no, as this causes many conflicts between the two Scrollviews. 

The suggested solution was to disable the inner Scrollview (this one does not need to scroll; they are using it more for the "grow" part).

The suggestion provided worked with the sample they attached, but it was not working on the actual application, and it was on them to validate what was different between them.

I can connect with the OP this week and find a solution for their application. 

Let's pair!

**Comment by NerijusN** [comment]
Yes it is still a problem, please observe in attached video, how underline appears:
(attn under Chart/Details)

https://github.com/unoplatform/uno/assets/10631521/5997971a-879b-4452-a641-24bae0101d9b

**Comment by kazo0** [comment]
@ajpinedam FYI

Notice the `Chart / Details / History / Time ...` tabbar area

**Comment by ajpinedam** [comment]
Hi @NerijusN thanks for the information and the video. Would it be possible sharing the XAML used to create that View, at least the one that correspond where the problem is seen (Tab Bar)? Also I am open for a pairing session so that we can find a solution to this issue. 

Please let me know

**Comment by NerijusN** [comment]
Can we move this discussion to private?

**Comment by MartinZikmund** [comment]
Only relevant to native
